tools/ci_build/github/linux/docker/scripts/install_deps.sh

```diff
@@ -117,6 +117,12 @@ ${PYTHON_EXE} -m pip install -r ${0/%install_deps\.sh/requirements\.txt}
 if [ $DEVICE_TYPE = "gpu" ]; then
   if [[ $INSTALL_DEPS_TRAINING = true ]]; then
     ${PYTHON_EXE} -m pip install -r ${0/%install_deps.sh/training\/requirements.txt}
+    # pip runs through the requirements.txt twice. First time downloading the packages and running each package's
+    # setup.py and the second time actually installing it.
+    # deepspeed has `import torch` in its setup.py. So having deepspeed as a part of requirements.txt even though torch
+    # package has already been collected results in an error.
+    # Work around is to install deepspeed separately after the requirements.txt is installed through secondary/requirements.txt
+    ${PYTHON_EXE} -m pip install -r ${0/%install_deps.sh/training\/secondary\/requirements.txt}
```

Comment:
This comment seems to be related to my previous question before, on why we need two requirements. Have you found an existing issue opened on deepspeed github repo? Isnt there any pip install flag to do download + install in a single shot?

Reply:
I don't believe there is any flag that would do that. I tried to search for issues on deepspeed's repository and was not able to find one. I will open an issue for them.

Reply:
I created an issue for deepspeed: https://github.com/microsoft/DeepSpeed/issues/663. I think until the issue is resolved, we can work with this approach.

Reply:
@baijumeswani this issue should be resolved once we merge this PR https://github.com/microsoft/DeepSpeed/pull/1166